.github/workflows/build-depends.yml

```diff
@@ -51,23 +51,38 @@ jobs:
             depends/sources
           key: depends-sources-${{ hashFiles('depends/packages/*') }}
           restore-keys: |
-            depends-sources-${{ hashFiles('depends/packages/*') }}
             depends-sources-
 
+      - name: Cache SDKs
+        uses: actions/cache@v4
+        if: inputs.build-target == 'mac'
+        with:
+          path: |
+            depends/sdk-sources
+            depends/SDKs
+          key: depends-sdks-${{ hashFiles('depends/hosts/darwin.mk') }}
+          restore-keys: |
+            depends-sdks-
+
       - name: Restore cached depends
         uses: actions/cache/restore@v4
         id: restore
         with:
           path: |
             depends/built
             depends/${{ steps.setup.outputs.HOST }}
-          key: ${{ runner.os }}-depends-${{ inputs.build-target }}-${{ steps.setup.outputs.DEP_HASH }}-${{ hashFiles('depends/packages/*') }}
+          key: depends-${{ hashFiles('contrib/containers/ci/Dockerfile', 'depends/packages/*') }}-${{ inputs.build-target }}-${{ steps.setup.outputs.DEP_HASH }}
           restore-keys: |
-            ${{ runner.os }}-depends-${{ inputs.build-target }}-${{ hashFiles('depends/packages/*') }}
-            ${{ runner.os }}-depends-${{ inputs.build-target }}
+            depends-${{ hashFiles('contrib/containers/ci/Dockerfile', 'depends/packages/*') }}-${{ inputs.build-target }}-
+            depends-${{ hashFiles('contrib/containers/ci/Dockerfile', 'depends/packages/*') }}-
```

Comment:
this feels like a not ideal order; if anything in depends/packages changes we will completely miss cache. Isn't it better to have it as `depends-${{ inputs.build-target }}-hashfiles-dep_hash`? this was if something in dep_hash or depends/packages changes we'll still get some cache restore

Reply:
Want to keep around `hashFiles('contrib/containers/ci/Dockerfile')` because we _want_ the cache to be invalidated if we bump the base image to avoid the need for manually clearing caches when we do bumps, can drop `depends/packages/*` though.

Reply:
I think we need to include depends/packages/* as well no? otherwise we will get exact cache hit, when we actually still need to build the depends? and then that means we won't save that new cache

Reply:
Reintroduced `${{ hashFiles('depends/packages/*') }}` in latest push